.github/workflows/flannel.yml

```diff
@@ -0,0 +1,18 @@
+name: flannel-images
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
Can we also do a docker file build on PR submission? That would help ensure the dockerfilie works

Reply:
We can replace `go` application to many Dockerfiles. Just for example:
- Dockerfile.0.12.0-servercore1809
- Dockerfile.0.13.0-servercore1809
- Dockerfile.0.12.0-nanoserver1809
- Dockerfile.0.13.0-nanoserver1809

When flannel rollout new version we just add new Dockerfiles

Reply:
Now we can build Dockerfile (/kubeadm/flannel/Dockerfile) with special argument to test it:
```
docker build -t sigwindowstools/flannel:0.13.0 --build-arg flannelVersion="0.13.0" .
```

Reply:
We don't need separate dockerfiles if you use ARG BASEIMAGE and a simple script or build template.  See The pause image for an example. We don't need quite as complex scenario but shows how to use a single build file for multiple OS versions

https://github.com/kubernetes/kubernetes/blob/9a57dc20a98dcc7a6b2bd1323d4e27f84477cd1b/build/pause/Dockerfile_windows

Reply:
Ok. So we can use Dockerfile from this PR to build different tags for different flannel versions because it based on ARG flannelVersion.

Or I didn't correctly understand your first comment from this thread.

Reply:
I think James is asking to update the triggers to look like

```yaml
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

So we can make sure PRs don't accidently break the docker builds.

Reply:
We probably shouldn't publish the image for PR builds tho

Reply:
If we add trigger on PR it may break (replace) stable docker tags.

Reply:
@jsturtevant I mean that `mcr.microsoft.com/windows/servercore:ltsc2019` replaces previous `mcr.microsoft.com/windows/servercore:ltsc2019` when some patch rolled out for it. And this images have no `latest` tag.

Dockerfile in this PR have arguments to build images for different version of flannel. `publish-flannel.go` use it to build tags for all flannel version >= 0.12.
- sigwindowstools/flannel:0.12.0
- sigwindowstools/flannel:0.13.0

This tags will replace previous when we change Dockerfile.

Reply:
> We probably shouldn't publish the image for PR builds tho

Agreed, we should only build (not push) the docker file to give a signal that it works.  

> replaces previous mcr.microsoft.com/windows/servercore:ltsc2019 when some patch rolled out for it. And this images have no latest tag.

`ltsc2019` is acting as the tag `latest` but there is still a tagged version and they both point to the same layer. Run docker `run --rm jess/reg:v0.15.8 tags mcr.microsoft.com/windows/servercore | sort -ur` and then inspect manifests like `docker manifest inspect --verbose mcr.microsoft.com/windows/servercore:ltsc2019` and `docker manifest inspect --verbose mcr.microsoft.com/windows/servercore:10.0.17763.1518` to see this.

We should tag a specific version for folks who want to use a specific version (this should really be done in production) and tag a `flannel:0.12.0` for usability.

Reply:
Ok. I will remove `dockerPush` from publish-flannel.go